proto/api/v2/compliance_results_service.proto

```diff
@@ -0,0 +1,142 @@
+syntax = "proto3";
+
+option go_package = "v2";
+option java_package = "io.stackrox.proto.api.v2";
+
+import "google/protobuf/timestamp.proto";
+import weak "google/api/annotations.proto";
+import "api/v2/common.proto";
+import "api/v2/search_query.proto";
+
+package v2;
+
+// RuleSeverity provides an enum of the rule severity values from the compliance operator
+enum RuleSeverity {
+    UNSET_RULE_SEVERITY   = 0;
+    UNKNOWN_RULE_SEVERITY = 1;
+    INFO_RULE_SEVERITY    = 2;
+    LOW_RULE_SEVERITY     = 3;
+    MEDIUM_RULE_SEVERITY  = 4;
+    HIGH_RULE_SEVERITY    = 5;
+}
+
+// ComplianceCheckResult provides the results of a compliance check against a rule
+message ComplianceCheckResult {
+    enum CheckStatus {
+        UNSET          = 0;
+        PASS           = 1;
+        FAIL           = 2;
+        ERROR          = 3;
+        INFO           = 4;
+        MANUAL         = 5;
+        NOT_APPLICABLE = 6;
+        INCONSISTENT   = 7;
+    }
+    string                     check_id     = 1;
+    string                     check_name   = 2;
+
+    // ClusterCheckStatus groups the result of the check by cluster
+    message ClusterCheckStatus {
+        string                     cluster_id   = 1;
+        CheckStatus                status       = 2;
+        google.protobuf.Timestamp  created_time = 3;
+    }
+    repeated ClusterCheckStatus clusters     = 3;
+    RuleSeverity                severity     = 4;
+    string                      description  = 5;
+    string                      instructions = 6;
+    string                      standard     = 7;
+    string                      control      = 8;
+}
+
+// ComplianceScanResult provides the results of a scan
+message ComplianceScanResult {
+    string                         scan_name     = 1;
+    string                         profile_name  = 2;
+    repeated ComplianceCheckResult check_results = 3;
+}
+
+// ComplianceScanResultSeverities groups the failures by severity
+message ComplianceScanResultSeverities {
+    int32 fail_count        = 1;
+    RuleSeverity severity   = 2;
+}
+
+// ComplianceScanStatsShim models statistics of checks for a given scan configuration
+message ComplianceScanStatsShim {
+    string                                  scan_name                = 1;
+    int32                                   number_of_checks         = 2;
+    int32                                   number_of_failing_checks = 3;
+    int32                                   number_of_passing_checks = 4;
+    repeated ComplianceScanResultSeverities severities               = 5;
+    google.protobuf.Timestamp               last_scan                = 6;
+}
+
+// ComplianceScanResultsOverview provides an overview and stats of the results of a scan configuration
+message ComplianceScanResultsOverview {
+    ComplianceScanStatsShim       scan_stats            = 1;
+    repeated string               profile_name          = 2;
+    repeated string               cluster_id            = 3;
+}
```

Comment:
I made a guess and concluded that this object would serve the ring pie charts. This object may make sense for the UI but not so much from API-only standpoint. How about we represent profiles and clusters as passing and failing where passing means all checks must have passed?

Reply:
Fair enough. We can expand the response to indicate pass/fail if it becomes desirable.